examples/custom-server/server.js

```diff
@@ -9,7 +8,7 @@ const handle = app.getRequestHandler()
 
 app.prepare().then(() => {
   createServer((req, res) => {
-    const parsedUrl = parse(req.url, true)
+    const parsedUrl = new URL(req.url, `http://${req.headers.host}`)
```

Comment:
Looking at this some more it seems to utilize `req.headers.host` which would allow for an attack by a user providing `host` on platforms where `host` is not overwritten.

Reply:
Interesting. This is verbatim the code from the Node.js core documentation.

I wonder if the node core team knows this?

Reply:
It's a pretty common attack vector afaik: acunetix.com/blog/articles/automated-detection-of-host-header-attacks/

Reply:
Overall I think we can use `http://a` or something like that as we don't use the host in `parsedUrl`

Reply:
Realized this change is being made in the custom-server example, but Next.js internally parses the url too if it's not provided, so that can be updated too.

Reply:
`new URL('/', 'https://a')` seems to work 👍